bundles/org.openhab.binding.rotel/src/main/java/org/openhab/binding/rotel/internal/communication/RotelFlagsMapping.java

```diff
@@ -23,174 +27,80 @@
-        return RotelFlagsMapping.isBitFlagOn(flags, zone3FlagNumber, zone3BitNumber);
+    public RotelFlagsMapping(List<RotelFlagInfo> infos) {
+        this.infos = new HashMap<>();
+        for (RotelFlagInfo info : infos) {
+            this.infos.put(info.infoType(), info);
+        }
     }
 
     /**
-     * Set the zone 3 indicator
-     *
-     * @param flags the table of flags
-     * @param on true to set the indicator to ON or false to set it to OFF
+     * Get the availability of the information
      *
-     * @throws RotelException in case the zone 3 indicator is undefined
+     * @return true if the information is available
      */
-    public void setZone3(byte[] flags, boolean on) throws RotelException {
-        RotelFlagsMapping.setBitFlag(flags, zone3FlagNumber, zone3BitNumber, on);
+    public boolean isInfoPresent(RotelFlagInfoType infoType) {
+        return infos.get(infoType) != null;
     }
 
     /**
-     * Get the zone 4 indicator
+     * Get the information
      *
+     * @param infoType the type of information
      * @param flags the table of flags
      *
-     * @return true if the indicator is ON in the flags or false if OFF
+     * @return true if the information is ON in the flags or false if OFF
      *
-     * @throws RotelException in case the zone 4 indicator is undefined
+     * @throws RotelException in case the information is undefined
      */
-    public boolean isZone4On(byte[] flags) throws RotelException {
-        return RotelFlagsMapping.isBitFlagOn(flags, zone4FlagNumber, zone4BitNumber);
+    public boolean isInfoOn(RotelFlagInfoType infoType, byte[] flags) throws RotelException {
+        RotelFlagInfo info = infos.get(infoType);
+        if (info == null || info.flagNumber() > flags.length) {
+            throw new RotelException("Info " + infoType.name() + " not available in flags");
+        }
+        return RotelFlagsMapping.isBitFlagOn(flags, info.flagNumber(), info.bitNumber());
     }
 
     /**
-     * Set the zone 4 indicator
+     * Set the information
      *
+     * @param infoType the type of information
      * @param flags the table of flags
-     * @param on true to set the indicator to ON or false to set it to OFF
+     * @param on true to set the information to ON or false to set it to OFF
      *
-     * @throws RotelException in case the zone 4 indicator is undefined
+     * @throws RotelException in case the information is undefined
      */
-    public void setZone4(byte[] flags, boolean on) throws RotelException {
-        RotelFlagsMapping.setBitFlag(flags, zone4FlagNumber, zone4BitNumber, on);
+    public void setInfo(RotelFlagInfoType infoType, byte[] flags, boolean on) throws RotelException {
+        RotelFlagInfo info = infos.get(infoType);
+        if (info == null || info.flagNumber() > flags.length) {
+            throw new RotelException("Info " + infoType.name() + " not available in flags");
```

Comment:
Is this considered an exceptional situation? I'm asking because this exception seems currently ignored by the callers. If nothing special is needed, you could simply return here. You could also return a boolean value if some callers need to know whether the flag exists and was set, but this is otherwise not considered as an error.

Reply:
This method is only called by method setInfoInFlags in class RotelModel.
And setInfoInFlags is then only used by the simulator handler, something the user cannot activate. And the exception is then caught and always ignored by the simulator.
I can replace the exception by a boolean as returned value.